src/core/lib/slice/slice.cc

```diff
@@ -327,6 +341,15 @@ grpc_slice grpc_slice_sub_no_ref(grpc_slice source, size_t begin, size_t end) {
   return subset;
 }
 
+grpc_slice grpc_slice_sub_no_ref(grpc_slice source, size_t begin, size_t end) {
```

Comment:
Why do we need this method and the one below?

Reply:
grpc_slice_sub_no_ref is an internal API method, as is grpc_slice_sub.

I have added grpc_slice_sub_no_ref_internal as an in-core method that can be used instead, when we want to preserve type information so the appropriate fastpaths can be called. The thought is that any code inside core should only be using the internal methods eventually - so far, I've only converted the one caller that then passes the slice to grpc_mdelem_from_slices().